components/admin_console/schema_admin_settings.jsx

```diff
@@ -666,13 +688,144 @@ export default class SchemaAdminSettings extends AdminSettings {
         );
     }
 
+    closeTooltip = () => {
+        this.setState({errorTooltip: false});
+    }
+
+    openTooltip = (e) => {
+        const elm = e.currentTarget.querySelector('.control-label');
+        const isElipsis = elm.offsetWidth < elm.scrollWidth;
+        this.setState({errorTooltip: isElipsis});
+    }
+
+    doSubmit = (callback, getStateFromConfig) => {
+        this.setState({
+            saving: true,
+            serverError: null,
+        });
+
+        // clone config so that we aren't modifying data in the stores
+        let config = JSON.parse(JSON.stringify(this.props.config));
+        config = this.getConfigFromState(config);
+
+        saveConfig(
+            config,
+            (savedConfig) => {
+                this.setState(getStateFromConfig(savedConfig));
+
+                this.setState({
+                    saveNeeded: false,
+                    saving: false,
+                });
+
+                this.props.setNavigationBlocked(false);
+
+                if (callback) {
+                    callback();
+                }
+
+                if (this.handleSaved) {
+                    this.handleSaved(config);
+                }
+            },
+            (err) => {
+                this.setState({
+                    saving: false,
+                    serverError: err.message,
+                    serverErrorId: err.id,
+                });
+
+                if (callback) {
+                    callback();
+                }
+
+                if (this.handleSaved) {
+                    this.handleSaved(config);
+                }
+            }
+        );
+    };
+
+    static getConfigValue(config, path) {
+        const pathParts = path.split('.');
+
+        return pathParts.reduce((obj, pathPart) => {
+            if (!obj) {
+                return null;
+            }
+
+            return obj[pathPart];
+        }, config);
+    }
+
+    setConfigValue(config, path, value) {
+        function setValue(obj, pathParts) {
+            const part = pathParts[0];
+
+            if (pathParts.length === 1) {
+                obj[part] = value;
+            } else {
+                if (obj[part] == null) {
+                    obj[part] = {};
+                }
+
+                setValue(obj[part], pathParts.slice(1));
+            }
+        }
+
+        setValue(config, path.split('.'));
+    }
+
+    isSetByEnv = (path) => {
+        return Boolean(SchemaAdminSettings.getConfigValue(this.props.environmentConfig, path));
+    };
+
     render = () => {
         const schema = this.props.schema;
 
         if (schema && schema.component) {
             const CustomComponent = schema.component;
             return (<CustomComponent {...this.props}/>);
         }
-        return AdminSettings.prototype.render.call(this);
+        return (
```

Comment:
All this is copied from AdminSettings class to remove the relationship (parent/child)